hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterClientProtocol.java

```diff
@@ -411,9 +411,8 @@ public boolean setReplication(String src, short replication)
     RemoteMethod method = new RemoteMethod("setReplication",
         new Class<?>[] {String.class, short.class}, new RemoteParam(),
         replication);
-    if (rpcServer.isInvokeConcurrent(src)) {
-      return !rpcClient.invokeConcurrent(locations, method, Boolean.class)
-          .containsValue(false);
+    if (rpcServer.isPathAll(src)) {
```

Comment:
This change will affect the original logic.

Reply:
I think the logic of `setReplication` api should be  consistent with `delete` api.  This change will ok.

Reply:
And I have add `setReplication ` unit tests and It run well.

Reply:
> I think the logic of `setReplication` api should be consistent with `delete` api. This change will ok.

This will be a problem if the path is a mount point. Or you could roll back the change for `setReplication` and open a new JIRA for it. If only the `truncate` is changed, I will +1.

Reply:
Sorry for my late reply. I got sick last week.  I have roll back the change for `setReplicaion` and will open a new JIRA for it.

Reply:
Thanks @Neilxzn for updating this.